src/az390.java

```diff
@@ -1878,17 +1882,28 @@ else if (tz390.op_trace_type[index]==102)
                 case 12:
                     my_format="RI";                                                    // #500
                     my_hexop=my_hexop.substring(0,2)+"."+my_hexop.substring(2);        // #500
-                    if (tz390.op_name[index].equals("BRC"))
-                       {my_operands="M1,I2";                                           // #500
-                        }
-                    else
-                       {my_operands="R1,I2";                                           // #500
-                        }
+                    my_operands="R1,I2";                                               // #554
                     break;
                 case 13:
                     // Our op_code table has the last two nibbles swapped
                     my_format="RI";                                                    // #500
-                    my_hexop=my_hexop.substring(0,2)+"."+my_hexop.substring(3)+my_hexop.substring(2,3); // #500
+                    if (my_hexop.length() == 3) // mask excluded?                      // #554
+                       {my_hexop=my_hexop.substring(0,2)+"."+my_hexop.substring(2,3);  // #554
+                        short_op = 1;           // mark short opcode                   // #554
+                        }                                                              // #554
+                    else                        // mask included!                      // #554
+                       {my_hexop=my_hexop.substring(0,2)+my_hexop.substring(3)+my_hexop.substring(2,3); // #554
+                        short_op = 0;           // mark full opcode                    // #554
+                        }                                                              // #554
+                    if (tz390.op_trace_type[index]==130)                               // #554
+                       {if (short_op == 1)      // short opcode                        // #554
```

Comment:
Just a suggestion. Nothing wrong.

You could define short_op as a boolean and set to true or false instead of 1 or 0, replacing `if *short_op == 1)` with `if (short_op)`. If you want to be slick, replace the `if ... else ...` with one line
```
my_operands = (short_op ? "R1,R12" : "R12");
```
Or, just leave it as is. If you decide to change, there are a few more short_op usages below.

Reply:
John, I never thought of the ternary operator. I find it confusing. I guess that merely shows my lack of experience with Java and C/C++
I prefer to leave as is, although I agree that it looks better from a coding perspective.